app/javascript/controllers/top_controls_controller.js

```diff
@@ -185,4 +200,86 @@ export default class extends Controller {
   #isMobile() {
     return window.innerWidth < 640 // Tailwind's sm breakpoint
   }
+
+  initializeTashkeelToggle() {
+    if (this.hasContentTarget) {
+      // Store the original content if we have tashkeel toggle buttons
+      if (this.hasTashkeelToggleButtonTarget) {
+        this.originalContent = this.contentTarget.innerHTML
+        this.showingTashkeel = true
+      }
+    }
+  }
+
+  toggleTashkeel() {
+    if (!this.hasContentTarget || !this.originalContent) return
+
+    this.showingTashkeel = !this.showingTashkeel
+
+    if (this.showingTashkeel) {
+      // Show original text with tashkeel
+      this.contentTarget.innerHTML = this.originalContent
+    } else {
+      // Hide tashkeel by removing diacritics
+      const textWithoutTashkeel = this.removeTashkeel(this.originalContent)
+      this.contentTarget.innerHTML = textWithoutTashkeel
```

Comment:
No need for the intermediate variable, just assign the function returned value to the `innerHTML`.